### DISCUSSION TITLE
Refactoring

### DISCUSSION THREAD

**Discussion by erik121097** [EP1210/Budget_Fox#32] [General]
# Tasks

Below you can find a list of tasks that need to be completed.

**Comment by michael-mezgo** [comment]
- [x] Tastatur verdeckt Eingabefelder
- [x] Alert falsche Userdaten beim Login
- [x] Datepicker überarbeiten (Feld)
- [x] UI für Income und Expense hinzufügen
- [x] Auge Icon Passwort Feld

**Comment by michael-mezgo** [comment]
- [x] Income / Expense - Screen / ViewModel cast from string to int in setter

**Comment by michael-mezgo** [comment]
- [x] Input validation

**Reply by michael-mezgo** [reply]
Done for number fields by @elimei1

**Comment by erik121097** [comment]
- [x]  State of input fields in TransactionScreen is not reset after adding an income or expense or navigating back
